components/modal/src/modal/modal.test.js

```diff
@@ -1,8 +1,30 @@
-import { render, screen } from '@testing-library/react'
+import { render, screen, fireEvent } from '@testing-library/react'
 import React from 'react'
+import { CloseButton } from './close-button.js'
 import { Modal } from './modal.js'
 
 describe('Modal', () => {
+    describe('Accessibility', () => {
+        it('closes when ESC key is pressed', () => {
```

Comment:
As mentioned in the `Split Button`-PR I think it would make sense to add some test-cases with button component that also handle "ESC keys". Eg. Selects. And test that the modal is not closed if esc is pressed when that is open.